data/tutorials/lg_08_error_handling.md

```diff
@@ -125,67 +191,549 @@ let () =
+cases that are not bugs (i.e., not `assert false`, but network failures, keys
+not present, etc.) to return a more explicit type such as `'a option` or `('a,
+'b) result` (see next section).
 
-The Stdlib module contains the following type:
+### Naming Conventions
+
+There are two naming conventions to have two versions of the same partial
+function: one raising exception, the other returning an option. In the above
+examples, the convention of the standard library is to add an `_opt` suffix to
+name of the version of the function that returns an option instead of raising
+exceptions.
+```ocaml
+val find: ('a -> bool) -> 'a list -> 'a
+(** @raise Not_found *)
+val find_opt: ('a -> bool) -> 'a list -> 'a option
+```
+This is extracted from the `List` module of the standard library.
+
+However, some projects tend to avoid or reduce the usage of exceptions. In such
+a context, reversing the convention is a relatively common idiom. It is the
+version of the function that raises exceptions that is suffixed with `_exn`.
+Using the same functions, that would be the specification
+```ocaml
+val find_exn: ('a -> bool) -> 'a list -> 'a
+(** @raise Not_found *)
+val find: ('a -> bool) -> 'a list -> 'a option
+```
+### Composing Functions Returning Options
+
+The function `div_opt` can't raise exceptions. However, since it doesn't return
+a result of type `int`, it can't be used in place of an `int`. The same way
+OCaml doesn't
+[promote](https://en.wikipedia.org/wiki/Type_conversion#Type_promotion) integers
+into floats, it doesn't automatically converts `int option` into `int` or _vice
+versa_.
+
+```ocaml
+# 21 + Some 21;;
+Error: This expression has type 'a option
+       but an expression was expected of type int
+```
+
+In order to combine option values with other values, conversion functions are
+needed. Here are the functions provided by the `Option` module to extract the
+data contained in an option:
+```ocaml
+val get : 'a t -> 'a
+val value : 'a t -> default:'a -> 'a
+val fold : none:'a -> some:('b -> 'a) -> 'b t -> 'a
+```
+`get` returns the content or raises `Invalid_argument` if applied to `None`.
+`value` essentially behaves as `get`, except it must be called with a default
+value which will be returned of if applied to `None`. `fold` also needs to be
+passed a default value that is returned when called on `None`, but it also
+expects a function that will be applied to the content of the option, when not
+empty.
+
+As a remark, observe that `value` can be implemented using `fold`:
+```ocaml
+# let value ~default = Option.fold ~none:default ~some:Fun.id;;
+val value : default:'a -> 'a option -> 'a = <fun>
+# Option.value ~default:() None = value ~default:() None;;
+- : bool = true
+# Option.value ~default:() (Some ()) = value ~default:() (Some ());;
+- : bool = true
+```
+
+It is also possible to perform pattern matching on option values:
+```ocaml
+match opt with
+| None -> ...    (* Something *)
+| Some x -> ...  (* Something else *)
```

Comment:
Same as the other questions... is this complete or should the reader be able to run it with an output (or combine it with something else that is runable)?